### PR TITLE
Save guide star selected in explore

### PR DESCRIPTION
**Pull request by cquiroz**
In this PR we try to integrate saving the guide star name when we don't use the automatic selection (aka next/prev buttons)

Storing the name is fairly straightforward but using it back is more complicated. For explore it is not enough the have the name, we need to hydrate the name with the results of ags analysis.

Keeping client and backend in sync adds some complications especially as the backend in invalidating some correct cases. A story was added to address that.

Automatic angle flips is the `Allow 180 flip` mode are disabled for now

### CODE REVIEW

File: explore/src/main/scala/explore/tabs/ObsTabTiles.scala [hunk 1]
```diff
@@ -221,6 +224,42 @@ object ObsTabTiles:
         p.observation.model.get.observationTime
       ): (_, _, _, _, _, _, _, _, _, _, _) =>
         vizTime => IO(vizTime.getOrElse(Instant.now()))
+      // Store guide star selection in a view for fast local updates
+      // This is not the ideal place for this but we need to share the selected guide star
+      // across the configuration and target tile
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/tabs/ObsTabTiles.scala:229]: it is true, it was on AladinCell before

File: common/src/main/scala/explore/model/GuideStarSelection.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    analysis:        AgsAnalysis
+  ) extends GuideStarSelection
+      derives Eq
+
+  val Default: GuideStarSelection = AgsSelection(None)
+
+  extension (gs: GuideStarSelection)
+    def fold[A](f: AgsSelection => A, g: RemoteGSSelection => A, h: AgsOverride => A): A =
+      gs match
+        case a @ AgsSelection(_)  => f(a)
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply common/src/main/scala/explore/model/GuideStarSelection.scala:37]: just general confusion

File: explore/src/main/scala/explore/targeteditor/AladinCell.scala [hunk 3]
```diff
@@ -333,23 +355,35 @@ object AladinCell extends ModelOptics with AladinCommon:
...
                   // set the selected index to the first entry
                   _ <- {
                     val index      = 0.some.filter(_ => r.exists(_.nonEmpty))
                     val selectedGS = index.flatMap(i => r.flatMap(_.lift(i)))
-                    (selectedIndex.set(index) *>
-                      props.obsConf
-                        .flatMap(_.selectedGS)
-                        .map(_.set(selectedGS))
-                        .getOrEmpty).unlessA(agsOverride.get.value)
+                    Callback.log(
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/targeteditor/AladinCell.scala:371]: Nope, let's remove it

**Review by toddburnside — APPROVED** [review]
LGTM

**Pull request merged**